libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteMetadata.kt

```diff
@@ -15,15 +15,18 @@ import com.mapbox.navigation.core.directions.session.RoutesObserver
  * from the perspective of the primary route
  * @param infoFromFork information about the alternative route from the fork with the primary route, until the destination
  * @param infoFromStartOfPrimary summed up information about the alternative route by joining
- * @param alternativeId is an id of alternative route. New alternative routes which matches tracking alternatives have the same alternativeId.
  * the primary route's data until the deviation point with the alternative route's data from the deviation point
+ * @param alternativeId is an id of alternative route. New alternative routes which matches tracking alternatives have the same alternativeId.
  */
 class AlternativeRouteMetadata internal constructor(
     val navigationRoute: NavigationRoute,
     val forkIntersectionOfAlternativeRoute: AlternativeRouteIntersection,
     val forkIntersectionOfPrimaryRoute: AlternativeRouteIntersection,
     val infoFromFork: AlternativeRouteInfo,
     val infoFromStartOfPrimary: AlternativeRouteInfo,
+    @Deprecated(
+        message = "This value can change on each route reset and shouldn't be used."
+    )
     val alternativeId: Int,
```

Comment:
I'm noticing that the alternative ID changes even when switching between existing routes, and also I see that if 2 alternatives are delivered, they both can have an ID of `0`. This makes the field unusable so I opted to call this out explicitly.

cc @VysotskiVadim @averkhaturau

Reply:
Unfortunately there's no alternative. I think we can keep the experiment of whether users accept/decline an alternative but we won't be able to prevent an alternative from being re-proposed.